(#1567) Capture global and modules values for checkpoint json

kevinatkinson-pnnl opened the pull request:
#### What's this Pull Request do?
Add Globals and modules information to the generated checkpoint json file

Revises json objects to be ordered json.

Adds some validation/health check features for API and checkpoint

kevinatkinson-pnnl committed:
(#1567)  Add globals to the checkpoint json

kevinatkinson-pnnl committed:
(#1567) Add modules to checkpoint json

Add type parsing to global variables

afisher1 reviewed (changes requested):
instead of nlohmann::json as checkpoint json object type use nlohmann::ordered_json to maintain the various link list order in the output json file?

kevinatkinson-pnnl committed:
(#1567) Revise json to be ordered_json

Removes part of nlohman download that causes issues

Adds checkpoint validate features to API

kevinatkinson-pnnl committed:
Merge remote-tracking branch 'upstream/feature/1478' into feature/1567

kevinatkinson-pnnl commented:
Merged into 1611, closing this PR to avoid conflicts.
